<?xml version="1.0"?>
<launch>
    <arg name="launch_rviz" default="true"/>

    <!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
    <!-- Launch Ouster OS1 Driver -->
    <include file="$(find ouster_ros)/launch/sensor.launch">
        <arg name="viz" value="false"/>
        <arg name="ouster_ns" value="ouster"/>
        <arg name="sensor_hostname" value="os-122138000087.local"/>
        <arg name="lidar_mode" value="1024x10"/>
    </include>

    <!-- Launch FLIR Camera Driver -->
    <include file="$(find ptu_control)/launch/camera.launch">
        <arg name="rviz" value="false"/>
    </include>
    <node if="$(arg launch_rviz)" name="rviz" pkg="rviz" type="rviz" required="true" args="-d $(find thermal_mapping)/rviz/tpc.rviz"/>
    <!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->

    <node pkg="thermal_mapping" name="timestamp_synchronizer" type="timestamp_synchronizer.py" output="screen" launch-prefix="xterm -e"/>

    <remap from="/os_cloud_node/points" to="/robot/ouster/points"/>
    <param name="/LiDAR_pointcloud_topic" type="string" value= "/laser_cloud_flat" />
    <param name="/IMU_topic" type="string" value= "/robot/imu/data_raw" />
    <!-- <param name="/Image_topic" type="string" value= "/NotAvail" /> -->
    <param name="Image_topic" type="string" value="/camera/image_raw_sync"/>
    <!-- <param name="Image_topic" type="string" value="/robot/front_rgbd_camera/rgb/image_raw"/> -->
    <param name="map_output_dir" type="string" value="$(env HOME)/r3live_output" />
    <rosparam command="load" file="$(find thermal_mapping)/cfg/thermal_camera.yaml" />
    
    <!-- set LiDAR type as ouster-64 spining LiDAR -->
    <param name="/Lidar_front_end/lidar_type" type="int" value= "3" /> 
    <param name="/Lidar_front_end/point_step" type="int" value="1" />
    <param name="r3live_lio/lio_update_point_step" type="int" value="6" />
        
    <node pkg="r3live" type="r3live_LiDAR_front_end" name="r3live_LiDAR_front_end" output="screen" required="true" launch-prefix="xterm -e"/>
    <node pkg="r3live" type="r3live_mapping" name="r3live_mapping" output="screen" required="true" launch-prefix="xterm -e"/>
</launch>